httpsync/publisher.go

```diff
@@ -93,13 +92,20 @@ func (p *publisher) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		// serve the
 		p.rl.RLock()
 		defer p.rl.RUnlock()
-		out, err := json.Marshal(p.root.String())
+
+		if p.privKey == nil {
+			http.Error(w, "Server not initialized", http.StatusInternalServerError)
+			log.Errorw("Failed to load private key to sign head request")
+			return
+		}
+
+		marshalledMsg, err := NewEncodedSignedHead(p.root, p.privKey)
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusBadRequest)
+			http.Error(w, "Failed to encode", http.StatusInternalServerError)
 			log.Errorw("Failed to serve root", "err", err)
-		} else {
-			_, _ = w.Write(out)
 		}
```

Comment:
I think there needs to be a return before this line.

Reply:
good catch!